src/ansible_navigator/runner/base.py

```diff
@@ -0,0 +1,218 @@
+                                   path will be volume mounted within the execution enviornment
+            set_environment_variable([dict], optional): Dict of user requested envvars to set
+            pass_environment_variable([list], optional): List of user requested envvars to pass
+            rotate_artifacts ([int], optional): Keep at most n artifact directories
+            timeout ([int], optional): The timeout value in seconds that will be passed to either
+                                       ``pexpect`` of ``subprocess`` invocation
+                                       (based on ``runner_mode`` selected) while executing command.
+                                       It the timeout is triggered it will force cancel the
+                                       execution.
+        """
+        self._logger = logging.getLogger(__name__)
+
+        self._set_private_data_directory(private_data_dir)
+        self._ce = container_engine
+        self._ee = execution_environment
+        self._eei = execution_environment_image
+        self._navigator_mode = navigator_mode
+        self._set_environment_variable: Dict[str, Any] = (
+            set_environment_variable if isinstance(set_environment_variable, dict) else {}
+        )
+        self._pass_environment_variable: List[str] = (
+            pass_environment_variable if isinstance(pass_environment_variable, list) else []
+        )
+        self._host_cwd = host_cwd
+        self._rotate_artifacts = rotate_artifacts if isinstance(rotate_artifacts, int) else None
+        self._timeout = timeout if isinstance(timeout, int) else None
+        self.ansible_runner_instance: Runner
+        self.cancelled: bool = False
+        self.finished: bool = False
+        self.status: Optional[str] = None
+        self._runner_args: Dict = {}
+        self._runner_artifact_dir: Optional[str] = None
+        if self._ee:
+            self._runner_args.update(
+                {
+                    "container_image": self._eei,
+                    "process_isolation_executable": self._ce,
+                    "process_isolation": True,
+                    "container_volume_mounts": container_volume_mounts,
+                    "container_options": container_options,
+                    "container_workdir": container_workdir,
+                }
+            )
+        self._runner_args.update(
+            {
+                "private_data_dir": self._private_data_dir,
+                "json_mode": True,
+                "quiet": True,
+                "cancel_callback": self.runner_cancelled_callback,
+                "finished_callback": self.runner_finished_callback,
+                "artifacts_handler": self.runner_artifacts_handler,
+                "timeout": self._timeout,
+            }
+        )
+        if self._rotate_artifacts is not None:
+            self._runner_args.update({"rotate_artifacts": self._rotate_artifacts})
```

Comment:
No need to overcomplicate a single assignment here.
```suggestion
            self._runner_args["rotate_artifacts"] = self._rotate_artifacts
```

Reply:
agree, fixed in #630